sympy/functions/elementary/complexes.py

```diff
@@ -140,12 +166,30 @@ class im(Function):
     >>> from sympy.abc import x, y
     >>> im(2*E)
     0
-    >>> re(2*I + 17)
-    17
+    >>> im(2*I + 17)
+    2
     >>> im(x*I)
     re(x)
     >>> im(re(x) + y)
     im(y)
+    >>> im(x + y*I)
+    re(y) + im(x)
+    >>> im(2 + 3*I).as_real_imag()
+    (3, 0)
+    >>> im(21*I + 50).as_real_imag(deep=True)
```

Comment:
This doesn't really demonstrate deep=True. I don't see why that needs to be shown here anyway as this is the docstring for `im`.

Reply:
Removed.